### DISCUSSION TITLE
Specialized command - move check or any solution for my case?

### DISCUSSION THREAD

**Discussion by bobhenl** [Zrips/CMI#3920] [Feature Requests]
**Description of feature/improvement request:**

---
### Detailed explanation of what it should do, when and in what situation: ###
Hi, I'm using custom alias with multiple command alias to random teleport to worlds, like:
```
  priroda:
    Cmds:
    - asConsole! rtp [playerName] world
    - delay! 10
    - ifinworld:world! asConsole! cmi ctellraw [playerName] <T>&7Byl jsi přemístěn
      do přírody! Pro tipy &enajeď &7myší.</T><H>&7Než začneš stavět, udělej si nejdříve
      \n&7pomocí &e/res auto &7či &ekosti &7residenci. \n&7Těžit můžeš na &e/doly
      \n&7či &e/tezba&7.\n\n &3Klikni pro teleport do těžebního světa.</H><SC>/tezba</SC>
```
Which teleports you using /priroda to the main world. If the teleport is succesfull, it will send player message (there's delay 10 seconds, because rtp isn't instantly).

And now, the problem is, that it can't check if the command was executed succesfully. I'm using HyperDrive https://www.spigotmc.org/resources/%E2%96%BA%E2%96%BA%E2%96%BA-hyperdrive-%E2%96%BA%E2%96%BA%E2%96%BA-advanced-teleportation-plugin-%E2%96%BA%E2%96%BA%E2%96%BA.17184/ plugin for teleportation due to nice looking warps menu and options there, so I'm using also their RTP and TP systems on my Survival server (on Semi-Vanilla I normally have CMI teleportation system, so my Semi-Vanilla looks like this:
```
  priroda:
    Cmds:
    - rt [playerName] world
    - ifinworld:world! asConsole! cmi ctellraw [playerName] <T>&7Byl jsi přemístěn
      do přírody! Začni svou cestu!</T>
```
Here is random teleport instant, so without delay, but again problem, that I'm checking just if he already isn't in world... so if he teleports from Nether and teleportation fails... then it won't send message, BUT when he will execute it in main world and it fails... it still sends message, because it's world "world". So I need some extra check, something like if coords were changed when executed command, or if was command executed succesfully (but not sure if you can check it also from other plugins), or maybe just if player won't move and won't move during it, maybe something like !notmove, so if player didn't move after executing command for 10 seconds (first case), then it will send message (moving is disabling teleportation, so this could be solution, something like !notmove).

But if you will have better idea to check, if was player really changed his location or if was command succesfully, would be nice.

### Basic explanation as to why this feature is needed and/or useful to have: ###
So the main thing is to prevent sending message, when players are in specific world (now main world) and the teleportation is cancelled (via move), so it shouldn't send them message. Not sure if it could be done now, I already checked documentation of specialized command and didn't get solution, also asked on your Dis.

I'm getting reports from player's that they're getting this message even if random teleport was cancelled and they were already in main world.

**Comment by CloudeLecaw** [comment]
You should remove the Meassage and modify the rusltmeassage of your rtp inside your langfile. Because you allready have a check reacting meassage there. But you cant execute a command onnsuccess of an other command. This would need a extanstion remade for nearly all commands. And even then its just possible to add this is only for cmi.
So i recomment you the better and easier way to modify the teleport sucess meassage of the langfile for all. Randomteleports to your needs and leave the custom alais with just the teleport command.
The biggest Problem you have at the Moment with your way: you have a line by line interpreter. 
And there is nonway to ask/check for command sucess inside these line interpred all you can donis define requirements checks, and a command sucess is not a requirement like Player is in wirld, or had this level or name...

**Comment by bobhenl** [comment]
CloudeLecaw but the problem is that in lagfile I can't set world-specific messages, I have special for nether rtp, main world rtp, mining world rtp and end rtp... in every message I have written features of this world. It's also hover and click event. I can't do this in normal lang file, that's why I'm using ctellraw and custom aliases. Need one extra check for it to let it works, don't know what check, but as I told you now I can't make it simply in language files.

**Comment by CloudeLecaw** [comment]
I see so your using different messages for different worlds. Well then there is no solution at the Moment because there is no check that asked if a command was executed successfull

**Comment by bobhenl** [comment]
Yeah, and that's what I'm asking about... something that could check atleast if he changed location, moved or something like this.
